src/OrleansRuntime/Facet/GrainConstructorFacetArguementsFactory.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Orleans.Runtime
+{
+    internal class GrainConstructorFacetArguementsFactory
```

Comment:
The general recommendation is to limit type names to three words. Maybe GrainConstructorFacetArguementsFactory -> FacetArguementsFactory or at least GrainFacetArguementsFactory ?